### PR TITLE
Refactor network handling - Closes #3267

### LINKED ISSUE #3267: Refactor setting of network and general network handling
### Expected behavior
The way network should be handled in the state and in the api requests should be readable and debuggable, and should follow coding standards. 

### Actual behavior
The code is very convoluted and hard to follow.

### PR DESCRIPTION
**Pull request by UsamaHameed**
### What was the problem?
This PR resolves #3267

### How was it solved?
Refactored the code so that it is much cleaner and follows our coding standards

### How was it tested?
Existing unit and e2e should be enough

### CODE REVIEW

File: src/components/shared/navigationBars/topBar/networkName.js [hunk 1]
```diff
@@ -1,16 +1,15 @@
 import React from 'react';
+import networks, { networkKeys } from '../../../../constants/networks';
```
Comment (reyraa) [inline src/components/shared/navigationBars/topBar/networkName.js:2]:
```suggestion
import { networkKeys } from '../../../../constants/networks';
```

File: src/utils/getNetwork.js [hunk 2]
```diff
@@ -28,25 +29,25 @@ export const getNetworkIdentifier = (network) => {
...
 
 
 export const getNetworkNameBasedOnNethash = (network, token = 'LSK') => {
   let activeNetwork = network.name;
-  if (network.name === networks.customNode.name && token !== 'BTC') {
-    activeNetwork = network.networks[token].nethash === Lisk.constants.TESTNET_NETHASH
-      ? networks.testnet.name
-      : network.name;
+  const isCustomNode = network.name === networkKeys.customNode;
+  const isBtc = token === 'BTC';
```
Comment (isalga) [inline src/utils/getNetwork.js:42]:
```suggestion
  const isBtc = token === tokenMap.BTC.key';
```

**Review by isalga — CHANGES_REQUESTED** [review]

**Review by reyraa — APPROVED** [review]
Great job @UsamaHameed
